pkg/datarepair/checker/checker.go

```diff
@@ -182,7 +186,11 @@ func (checker *Checker) getMissingPieces(ctx context.Context, pieces []*pb.Remot
 	}
 
 	for i, node := range nodes {
-		if node == nil || !checker.overlay.IsOnline(node) || !checker.overlay.IsHealthy(node) {
+		if node == nil {
+			if _, err = checker.kad.FetchPeerIdentity(ctx, nodeIDs[i]); err != nil {
```

Comment:
why are we fetching peer identity is the node is nil? just curious

Reply:
a nil node means that the overlay doesn't have a record at all.  Its likely perfectly fine, but for whatever reason we don't have a record of it.  I'm still not sure how this is even possible, given that if pointerdb knows of a node, I would think that overlay should know of the node.  
Nevertheless, we are getting nils from overlay.  :/